include/hybridmap/HybridNeutralAtomMapper.hpp

```diff
@@ -0,0 +1,534 @@
+                             InitialCoordinateMapping initialCoordinateMapping =
+                                 InitialCoordinateMapping::Trivial,
+                             const MapperParameters& p = MapperParameters())
+      : arch(arch), mappedQc(arch.getNpositions()),
+        mappedQcAOD(arch.getNpositions()), scheduler(arch), parameters(p),
+        hardwareQubits(arch, initialCoordinateMapping, parameters.seed) {
+    // need at least on free coordinate to shuttle
+    if (arch.getNpositions() - arch.getNqubits() < 1) {
+      this->parameters.gateWeight      = 1;
+      this->parameters.shuttlingWeight = 0;
+    }
+  };
+
+  /**
+   * @brief Sets the runtime parameters of the mapper.
+   * @param p The runtime parameters of the mapper
+   */
+  void setParameters(const MapperParameters& p) {
+    this->parameters = p;
+    if (arch.getNpositions() - arch.getNqubits() < 1) {
+      this->parameters.gateWeight      = 1;
+      this->parameters.shuttlingWeight = 0;
+    }
+  }
+
+  // Methods
+  /**
+   * @brief Maps the given quantum circuit to the given architecture.
+   * @details The mapping has following important parts:
+   * - initial mapping: The initial mapping of the circuit qubits to the
+   * hardware qubits.
+   * - layer creation: The creation of the front and lookahead layers, done one
+   * the fly and taking into account basic commutation rules.
+   * - estimation: The estimation of the number of swap gates and moves needed
+   * to execute a given gate and the decision which technique is better.
+   * - gate based mapping: SABRE based algorithm to choose the bast swap for the
+   * given layers.
+   * - shuttling based mapping: Computing and evaluation of possible moves and
+   * choosing best.
+   * - multi-qubit-gates: Additional steps and checks to bring multiple qubits
+   * together.
+   * -> Final circuit contains abstract SWAP gates and MOVE operations, which
+   * need to be decomposed using convertToAod method.
+   *
+   * @param qc The quantum circuit to be mapped
+   * @param initialMapping The initial mapping of the circuit qubits to the
+   * hardware qubits
+   * @param verbose If true, prints additional information
+   * @return The mapped quantum circuit with abstract SWAP gates and MOVE
+   * operations
+   */
+  QuantumComputation map(qc::QuantumComputation& qc,
+                         InitialMapping initialMapping, bool verbose = true);
+
+  /**
+   * @brief Maps the given quantum circuit to the given architecture and
+   * converts it to the AOD level. Used for Python bindings.
+   * @param qc  The quantum circuit to be mapped
+   * @param initialMapping The initial mapping of the circuit qubits to the
+   * hardware qubits
+   */
+  [[maybe_unused]] void mapPy(qc::QuantumComputation& qc,
+                              InitialMapping initialMapping, bool printInfo) {
+    map(qc, initialMapping, verbose = printInfo);
+    convertToAod(this->mappedQc);
+  }
+
```

Comment:
If only used for the python bindings, then consider moving this function there. Both of the methods used are public methods.

Reply:
The function might also be interesting for non binding things, so I might leave it here. I renamed it so it's use becomes more clear.